src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -36,9 +35,7 @@ Status CoreWorkerDirectActorTaskSubmitter::SubmitTask(
 
   std::unique_lock<std::mutex> guard(mutex_);
 
-  auto iter = actor_states_.find(actor_id);
-  if (iter == actor_states_.end() ||
-      iter->second.state_ == ActorTableData::RECONSTRUCTING) {
+  if (rpc_client == nullptr) {
```

Comment:
This is super weird to be passing in null clients. Instead of the current breakdown of SubmitTask(option<client>), SendPendingTasks(), and FailPendingTasks(), how about:

1. Keep the signature of SubmitTask the same as before.
2. Keep the internal pointer to the current RPC client.
3. Add a OnActorLocationUpdate(client) method to this class.

That way, the logic of this class becomes a lot more self contained, and core_worker.cc just needs to post the location update to the transport when it changes.